Provide tutorial for OIDC Authentication with AWS ALB on Kubeflow

PatrickXYS opened the pull request:
**Which issue is resolved by this Pull Request:**
Resolves #1672

**Description of your changes:**
We've already provided a Cognito Authentication tutorial for our users, but some of aws users want to use OIDC with ALB for authentication. Thus, we provide a tutorial of setting up OIDC authentication with Github Login with ALB on Kubeflow. 

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kubeflow/website/1780)
<!-- Reviewable:end -->

PatrickXYS commented:
/WIP

PatrickXYS commented:
/assign @Jeffwan

PatrickXYS commented:
It's still under development. The reason I want to create the PR is keeping you updated. @Jeffwan

Jeffwan commented on content/docs/aws/authentication-oidc.md:
```diff
@@ -0,0 +1,129 @@
+
+The details of instructions are listed below:
+
+### ALB Setup
```
Hi @PatrickXYS,

I think you misunderstand what I want you to help. We want user to use kfctl to setup OIDC, similarly like what we did for cognito, that's why I give you instructions here 

```
oidc:
    oidcIssuer: https://kubeflow.auth0.com/
    oidcAuthorizationEndpoint: https://kubeflow.auth0.com/authorize
    oidcTokenEndpoint: https://kubeflow.auth0.com/oauth/token
    oidcUserInfoEndpoint: https://kubeflow.auth0.com/userinfo

    # this should be ACM's certArn rather than cognito user pool arn
    certArn: arn:aws:acm:us-west-2:xxxx:certificate/c7a9eb05-0975-4fc8-bb79xxxxx97e4031c
    oAuthClientId: yours
    oAuthClientSecret: yours
```

The reason I ask you to do it manually is because kfctl is not ready at that time.

PatrickXYS replied:
Yes, it's my misunderstanding. And it's the confusion that I want to sync with you on Friday, to make it more clear. I'm working on it right now.

Jeffwan commented on content/docs/aws/authentication-oidc.md:
```diff
@@ -0,0 +1,129 @@
+## Enable TLS and Authentication
+[OIDC](https://openid.net/connect/) allows clients to verify the identity of the End-User based on the authentication performed by an Authorization Server. OIDC allows clients of all types, including Web-based, mobile, and JavaScript clients, to request and receive information about authenticated sessions and end-users. For this tutorial, we take Github Login as an example.
+
+In order to authenticate and manage users for Kubeflow, let's first connect Auth0 app to Github. You can follow instructions [here](https://auth0.com/docs/connections/social/github).
```
You probably want to address `Auth0` and `Github` is the solution you pick for the tutorial, user can use any OIDC compatible solutions.

PatrickXYS replied:
Sure, users are free to use any OIDC compatible solutions. And I mentioned "we take Github Login as an example". And thank you for your reminding, I'll add some words in the doc.

Jeffwan commented on content/docs/aws/authentication-oidc.md:
```diff
@@ -0,0 +1,129 @@
+After modification, click **NEXT** button. Then you can set up your personal information there and click **CREATE ACCOUNT** button.
+ 
+After log into Auth0 Dashboard, select **Connections > Social**, then choose Github. Copy the `Client ID` and `Client Secret` from the `Developer Applications` of your app on Github into the fields on this page on Auth0.
+
```
Can you adjust the sequence? it would be better to register auth0 because tenant name may not be available.

PatrickXYS replied:
No problem! Will fix it now

PatrickXYS commented:
/WIP

Jeffwan commented:
@PatrickXYS You can use WIP in the PR title in the future, I don't suggest you use it in commit log, and also make some other commits meaningful. One example is the new commit `Fix issue`.

PatrickXYS commented:
@Jeffwan  Thank you for your suggestion, and for new added commit, they exist temporarily and I'll squash them after finished development. And I'll make new commit more meaningful.

Jeffwan commented:
/cc @sarahmaddox

Jeffwan commented:
/ok-to-test

Jeffwan commented:
It looks good to me on the technical side. 

/lgtm

sarahmaddox commented:
Preview: https://deploy-preview-1780--competent-brattain-de2d6d.netlify.com/docs/aws/authentication-oidc/

/assign

sarahmaddox commented on content/docs/aws/authentication-oidc.md:
```diff
@@ -0,0 +1,128 @@
+weight = 90
++++
+
+This section shows the how to setup Kubeflow with authentication and authorization support through OIDC in Amazon Web Services (AWS). Similar to [Cognito tutorial](https://www.kubeflow.org/docs/aws/authentication/).
```
For references to other docs on the Kubeflow website, we use relative URLs. This is necessary for the doc versioning, where we have a different website for each major version of Kubeflow. The relative URL keeps the user within the same website.

Change to:
```
[Cognito tutorial](/docs/aws/authentication/)
```

PatrickXYS replied:
Great point! I'll address it in the new commit.

PatrickXYS commented on content/docs/aws/authentication-oidc.md:
```diff
@@ -0,0 +1,128 @@
+
+### 5. Deploy Kubeflow
+
+Download [kfctl_aws_cognito.v1.0.0.yaml](https://raw.githubusercontent.com/kubeflow/manifests/v1.0-branch/kfdef/kfctl_aws_cognito.v1.0.0.yaml). Before you `kfctl apply -V -f kfctl_aws_cognito.v1.0.0.yaml`, please update **spec** and **repos** fields in your Kubeflow configuration file at `kfctl_aws_cognito.v1.0.0.yaml`, so that it looks like this:
```
Yes, make the shortcodes is an efficient way for us to make future change. Thank you for your guidance!

PatrickXYS commented:
deploy/netlify

PatrickXYS commented:
Preview: https://5e6ac7630f252a0008e6691a--competent-brattain-de2d6d.netlify.com/

@sarahmaddox In the latest commit, I addressed your comments and you could check the preview. Thank you for guidance!

sarahmaddox commented:
Thanks @PatrickXYS!
/lgtm
/approve
